### ISSUE TITLE
service linking to INGRESS network 

### ISSUE THREAD

**Issue by xxradar** [moby/moby#29400]
hi ,

 when creating services like:

```
docker service create --constraint node.role==worker \
--constraint node.labels.service==stress \
--network stress-net-3 \
--name stress_ab \
--replicas 5 \
dockersec/ab -c 1 -n 1000 http://10.90.122.101/
```
every-time the container exists, a new container is lauched (as expected).

Although the service is linked to a swarm global overlay stress-net-3, it seems to try also a IPAM request for INGRESS network although there is no listening socket.

```
time="2016-11-02T23:03:44.796307364Z" level=error msg="task allocation failure" error="failed during networktask allocation for task 7qeighnmzapdtx7lk6iyjokww: failed to allocate network IP for task 7qeighnmzapdtx7lk6iyjokww network 3kwq0v051aim4ff2vzwmp1rlh: could not allocate IP from IPAM: Address already in use"
```

```
docker network ls
3kwq0v051aim ingress overlay swarm
```

these service spinning up / down seem to exhaust the IP address space of INGRESS.

Can we avoid a service linking to INGRESS ?

**Comment by thaJeztah** [comment]
ping @mavenugo PTAL

**Comment by aaronlehmann** [comment]
ping @aboch

**Comment by aboch** [comment]
@xxradar 
You are right, no requests for an IP on the ingress network should be made for tasks belonging to a service with no published ports. And I do not see this happening with a recent daemon.

What docker version are you using ?

Also, please provide the docker version, docker info and some daemon logs when the tasks are being re-spawned.

Thanks.

**Comment by mustafayildirim** [comment]
I have same problem.
`
172.17.0.1:~ # docker version
Client:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:
 OS/Arch:      linux/amd64
` 

`172.17.0.1:~ # docker info
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 1
Server Version: 1.12.2
Storage Driver: devicemapper
 Pool Name: docker-253:1-1175382-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 498.7 MB
 Data Space Total: 107.4 GB
 Data Space Available: 25.79 GB
 Metadata Space Used: 1.397 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.146 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay host bridge null
Swarm: active
 NodeID: 0xd9pv31tstr04e2et2a8hyue
 Is Manager: true
 ClusterID: 5xq7z6udtxk13iqlm2m9hg488
 Managers: 3
 Nodes: 9
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.49.66.110
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-327.36.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.632 GiB
Name: swarm-manager1
ID: HMX3:RVAY:GXT2:VGSD:QD25:RR5N:SRRG:BUJF:3IXN:XYF4:JSJU:K3AE
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8`


`172.17.0.1:~ # docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
c54d6ffdb9ed        bridge              bridge              local
857e1408f290        docker_gwbridge     bridge              local
6a9ba930c97a        host                host                local
5m8nkf1f4qdg        ingress             overlay             swarm
95b590d60d70        none                null                local`
And attach daemon-log
[damon-log.txt](https://github.com/docker/docker/files/747085/damon-log.txt)

**Comment by thaJeztah** [comment]
@mustafayildirim can you try updating to the current version of docker (either 1.12.6 if you want to stay on 1.12, or 1.13.0)

**Comment by mustafayildirim** [comment]
@thaJeztah I updated 12.6.1 but same problem, I have 3 master and 6 worker also.
Sometimes http://manager1:port1 is down and other two managers is up and then http://manager2:port1 is down. I am comfused. I know It is about routing mash. But I can't find the real issue.
I did these;
- Drain manager1
- Stop manager1 docker
- Clean IP tables
- Reboot manager1
- Active manager1

Also I have 3 others swarm cluster but none of them giving like that problem.
This cluster has many network requests. Maybe this is the problem.

**Comment by thaJeztah** [comment]
@mustafayildirim does that task also belong to a service that doesn't expose ports? Is network `5m8nkf1f4qdgs92p7x4ej4wtg` your ingress network?

w.r.t. managers being down; this looks a bit odd; I'll have to check the exact reason this message is printed, but to get some info; does manager1 have multiple network interfaces, and was `--advertise-addr` used when initialising the swarm? Does it have a dynamic IP-address?

```
Feb  1 14:32:11 swarm-manager1 dockerd: time="2017-02-01T14:32:11.958586110+03:00" level=warning msg="2017/02/01 14:32:11 [WARN] memberlist: Got ping for unexpected node swarm-manager1-5dcf4a2e0941 from=10.49.66.114:42966\n"
Feb  1 14:32:11 swarm-manager1 dockerd: time="2017-02-01T14:32:11.958923824+03:00" level=warning msg="2017/02/01 14:32:11 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.112:7946\n"
Feb  1 14:32:11 swarm-manager1 dockerd: time="2017-02-01T14:32:11.959001256+03:00" level=warning msg="2017/02/01 14:32:11 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.115:7946\n"
Feb  1 14:32:11 swarm-manager1 dockerd: time="2017-02-01T14:32:11.959063539+03:00" level=warning msg="2017/02/01 14:32:11 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.110:7946\n"
Feb  1 14:32:12 swarm-manager1 dockerd: time="2017-02-01T14:32:12+03:00" level=info msg="Firewalld running: false"
Feb  1 14:32:12 swarm-manager1 dockerd: time="2017-02-01T14:32:12+03:00" level=info msg="Firewalld running: false"
Feb  1 14:32:14 swarm-manager1 dockerd: time="2017-02-01T14:32:14.757177346+03:00" level=warning msg="2017/02/01 14:32:14 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.116:7946\n"
Feb  1 14:32:14 swarm-manager1 dockerd: time="2017-02-01T14:32:14.832069716+03:00" level=warning msg="2017/02/01 14:32:14 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.111:7946\n"
Feb  1 14:32:15 swarm-manager1 dockerd: time="2017-02-01T14:32:15.257718717+03:00" level=warning msg="2017/02/01 14:32:15 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.114:7946\n"
Feb  1 14:32:15 swarm-manager1 dockerd: time="2017-02-01T14:32:15.258021843+03:00" level=warning msg="2017/02/01 14:32:15 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.111:7946\n"
Feb  1 14:32:15 swarm-manager1 dockerd: time="2017-02-01T14:32:15.258176705+03:00" level=warning msg="2017/02/01 14:32:15 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.113:7946\n"
Feb  1 14:32:15 swarm-manager1 dockerd: time="2017-02-01T14:32:15.258464066+03:00" level=warning msg="2017/02/01 14:32:15 [WARN] memberlist: Got ping for unexpected node swarm-manager1-5dcf4a2e0941 from=10.49.66.116:37484\n"
Feb  1 14:32:15 swarm-manager1 dockerd: time="2017-02-01T14:32:15.332925794+03:00" level=warning msg="2017/02/01 14:32:15 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.110:7946\n"
Feb  1 14:32:15 swarm-manager1 dockerd: time="2017-02-01T14:32:15.333026253+03:00" level=warning msg="2017/02/01 14:32:15 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.113:7946\n"
Feb  1 14:32:15 swarm-manager1 dockerd: time="2017-02-01T14:32:15.333620231+03:00" level=warning msg="2017/02/01 14:32:15 [WARN] memberlist: Got ping for unexpected node swarm-manager1-5dcf4a2e0941 from=10.49.66.111:45674\n"
Feb  1 14:32:15 swarm-manager1 dockerd: time="2017-02-01T14:32:15.333724687+03:00" level=warning msg="2017/02/01 14:32:15 [WARN] memberlist: Got ping for unexpected node 'swarm-manager1-5dcf4a2e0941' from=10.49.66.116:7946\n"
```

**Comment by xxradar** [comment]
Hi,
Any chance one of your managers changed IP Address ?
https://www.peerlyst.com/posts/my-way-of-recovering-from-a-docker-swarm-meltdown-effectively-philippe-bogaerts <https://www.peerlyst.com/posts/my-way-of-recovering-from-a-docker-swarm-meltdown-effectively-philippe-bogaerts>

Kind regards,


I got a similar issue after problems in my (test) lab.  I just removed the manager and added it again.

**Comment by mustafayildirim** [comment]
@thaJeztah sure `5m8nkf1f4qdgs92p7x4ej4wtg` is ingress network.
```
172.17.0.1
172.18.0.1:~ # docker network inspect 5m8nkf1f4qdgs92p7x4ej4wtg
[
    {
        "Name": "ingress",
        "Id": "5m8nkf1f4qdgs92p7x4ej4wtg",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.255.0.0/16",
                    "Gateway": "10.255.0.1"
                }
            ]
        },
        "Internal": false,
        "Containers": {
            "ingress-sbox": {
                "Name": "ingress-endpoint",
                "EndpointID": "72f20f06696946dacd6057540459e5a1a65804073998f9bd0c527e1e23256ba0",
                "MacAddress": "02:42:0a:ff:00:03",
                "IPv4Address": "10.255.0.3/16",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "256"
        },
        "Labels": {}
    }
]
```

Nope when the service was removed, the port stuck. Docker still listens the port. 
I have one network interface at manager1. IP address is static. I cant remember ```--advertise-addr``` is used   when the swarm cluster initializing.

```
root@swarm-manager1_10.49.66.110
172.17.0.1
172.18.0.1:~ # ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eno16777984: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
    link/ether 00:50:56:a7:50:1a brd ff:ff:ff:ff:ff:ff
    inet 10.49.66.110/24 brd 10.49.66.255 scope global eno16777984
       valid_lft forever preferred_lft forever
3: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN
    link/ether 02:42:ca:06:e0:fb brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 scope global docker0
       valid_lft forever preferred_lft forever
8: docker_gwbridge: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP
    link/ether 02:42:c5:38:16:ff brd ff:ff:ff:ff:ff:ff
    inet 172.18.0.1/16 scope global docker_gwbridge
       valid_lft forever preferred_lft forever
10: veth8329bb7@if9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP
    link/ether 96:2a:f4:88:0e:5d brd ff:ff:ff:ff:ff:ff link-netnsid 1
```

**Comment by mustafayildirim** [comment]
@thaJeztah 
I say something interesting.
Not being a service that listens port but docker still listens that port.
I saw when I used `netstat -antlp`
```
172.17.0.1
172.18.0.1:~ # netstat -antlp | grep 740
tcp6       0      0 :::8740                 :::*                    LISTEN      2468/dockerd
```
The port: 8740 
And than. I created nginx service like that.
- `docker service create --name nginx nginx:alpine`
- `docker service update --publish-add 8740:80 nginx`
- `docker service rm nginx`

There is no port on 8740 listening.
Something weird.
